src/os/bluestore/BlueStore.cc

```diff
@@ -6361,6 +6370,12 @@ void BlueStore::_txc_state_proc(TransContext *txc)
 	return;
       }
       txc->state = TransContext::STATE_FINISHING;
+      if (txc->kv_submitted_sync && !txc->wal_txn) {
+	std::lock_guard<std::mutex> l(kv_lock);
+	kv_queue.push_back(txc);
+	kv_cond.notify_one();
+	return;
+      }
```

Comment:
I added this condition to avoid trying to take osr->qlock in _txc_finish method several times, which is conflict with _txc_finish_io method taking osr->qlock.
To solve this problem, calling _txc_finish method is delegated to kv_sync_thread In case of metadata only I/O.